com.teamscale.polarion.plugin/src/com/teamscale/polarion/plugin/WorkItemUpdatesServlet.java

```diff
@@ -0,0 +1,769 @@
+        fieldChange = new WorkItemFieldDiff(fieldDiff.getFieldName());
+        fieldChange.setElementsAdded(Utils.castApprovalsToStrList(added));
+      } else if (!Utils.isCollectionLinkedWorkItemStructList(added)) {
+        fieldChange = new WorkItemFieldDiff(fieldDiff.getFieldName());
+        try {
+          fieldChange.setElementsAdded(Utils.castCollectionToStrList((List<IPObject>) added));
+        } catch (ClassCastException ex) {
+          // For now, when an added element/value is not among the ones we're supposed
+          // to support in Teamscale, we simply add as an empty string array.
+          // Alternatively, we could ignore the field as a change
+          // (skip the field from the json output)
+          fieldChange.setElementsAdded(new ArrayList<String>());
+        }
+      }
+      if (fieldChange != null) {
+        workItemChange.addFieldChange(fieldChange);
+      }
+    }
+    if (removed != null && !removed.isEmpty()) {
+      WorkItemFieldDiff fieldChange = null;
+      if (Utils.isCollectionHyperlinkStructList(removed)) {
+        fieldChange = new WorkItemFieldDiff(fieldDiff.getFieldName());
+        fieldChange.setElementsRemoved(Utils.castHyperlinksToStrList(removed));
+      } else if (includeLinkRoles != null && Utils.isCollectionLinkedWorkItemStructList(removed)) {
+        String linkRoleId =
+            ((ILinkedWorkItemStruct) removed.iterator().next()).getLinkRole().getId();
+        if (Arrays.stream(includeLinkRoles).anyMatch(linkRoleId::equals)
+            && fieldDiff.getFieldName().equals(Utils.LINKED_WORK_ITEMS_FIELD_NAME)) {
+          fieldChange =
+              new LinkFieldDiff(
+                  fieldDiff.getFieldName(),
+                  ((ILinkedWorkItemStruct) removed.iterator().next()).getLinkRole().getId());
+          fieldChange.setElementsRemoved(Utils.castLinkedWorkItemsToStrList(removed));
+          updateOppositeLinksMap(workItemId, workItemChange.getRevision(), removed, false);
+        }
+      } else if (Utils.isCollectionApprovalStructList(removed)) {
+        fieldChange = new WorkItemFieldDiff(fieldDiff.getFieldName());
+        fieldChange.setElementsRemoved(Utils.castApprovalsToStrList(removed));
+      } else if (!Utils.isCollectionLinkedWorkItemStructList(removed)) {
+        fieldChange = new WorkItemFieldDiff(fieldDiff.getFieldName());
+        try {
+          fieldChange.setElementsRemoved(Utils.castCollectionToStrList((List<IPObject>) removed));
+        } catch (ClassCastException ex) {
+          // For now, when a removed element/value is not among the ones we're supposed
+          // to support in Teamscale, we simply add as an empty string array.
+          // Alternatively, we could ignore the field as a change
+          // (skip the field from the output)
+          fieldChange.setElementsRemoved(new ArrayList<String>());
+        }
+      }
+      if (fieldChange != null) {
+        workItemChange.addFieldChange(fieldChange);
+      }
```

Comment:
This is the exact same block as the above one, right? Just `setElementsRemoved` is called, instead. Then I would propose to extract this into its own method and provide the added/removed behavior either as functional parameter or as boolean and the method then decides internally whether add/remove should be called

Reply:
Yes, I always had this in mind but was lazy to refactor, but I just did it.

Done